pkg/controllers/allocation/controller.go

```
@@ -188,15 +212,15 @@ func (c *Controller) podToProvisioner(o client.Object) (requests []reconcile.Req
 
 // getProvisionerFor retrieves the provisioner responsible for the pod
 func (c *Controller) getProvisionerFor(ctx context.Context, p *v1.Pod) (*v1alpha3.Provisioner, error) {
-	provisionerKey := client.ObjectKey{Name: "default"}
+	if err := c.Filter.isUnschedulable(p); err != nil {
+		return nil, fmt.Errorf("pod is not eligible to be associated with a provisioner")
```

Can we propagate the error message from the filter.